client/src/elements/Prompt.jsx

```diff
@@ -21,33 +32,57 @@ export function Prompt({ file, name, shared }) {
   const progressLabel = player.get("progressLabel");
   const { text: promptString, error: fetchError } = useText({ file });
   const permalink = usePermalink(file);
+
   const [responses, setResponses] = React.useState([]);
 
   if (fetchError) {
     return <p>Error loading prompt, retrying...</p>;
   }
-  if (!promptString) return <p>Loading prompt...</p>;
+
+  if (!promptString) return <p>Prompt file is empty</p>;
 
   // Parse the prompt string into its sections
-  const sectionRegex = /---\n/g;
   const [, metaDataString, prompt, responseString] =
-    promptString.split(sectionRegex);
+    promptString.split(/^-{3,}$/gm);
+
+  // Check if each section is present
+  if (metaDataString === undefined || prompt === undefined || responseString === undefined) {
+    return (
+      <>
+        <p>Error: at least one of metadata, prompt, or response sections are missing. Please verify the formatting of the prompt markdown file.</p>
+      </>);
+  }
+
+  let metaData;
+
+  // Handling for YAMLException from loadYaml
+  try {
+    metaData = loadYaml(metaDataString);
+  } catch (e) {
+    console.log(e);
+    return (
+      <>
+        <p>Error in metadata section of prompt. Please verify that metadata section is correctly formatted as first section of markdown file.</p>
```

Comment:
In theory, this won't get rendered in the actual experiment, right? because we check for all of this on batch creation on the server side?